hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaObserverChore.java

```diff
@@ -566,6 +566,13 @@ void setTableQuotaSnapshot(TableName table, SpaceQuotaSnapshot snapshot) {
     this.tableQuotaSnapshots.put(table, snapshot);
   }
 
+  /**
+   * Removes the quota state for the given table.
+   */
+  void removeTableQuotasnapshot(TableName table) {
+    this.tableQuotaSnapshots.remove(table);
```

Comment:
I think you need some synchronization around modifications to this data structure. If I understand correctly, the Chore will be run in it's own thread, while quota changes would be dispatched from an IPC handler thread.

Reply:
Yes, sure.